### ISSUE TITLE
Incoming call notifications are broken on Android 12/Galaxy S22

### ISSUE THREAD

**Issue by eugeneseppel** [signalapp/Signal-Android#13002]
- [X] I have searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [X] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
After upgrading to a new Galaxy S22 with Android 12 I found out several issues with incoming call Notifications that basically makes Signal unusable. These issues did not exist on Android 8.x and Galaxy A5/A3.

In case of incoming call Signal messenger doesn't appear on the screen (Usual full screen window with contact picture, two Green/Red buttons to accept or decline the call) The only thing that appears -- small Android alert notification with two buttons (Decline, Answer). Only after clicking on this Alert usual full-screen window appears). The problem with this alert lasts only for several seconds, after that disappears as well as sound and vibration. So it's too easy to miss the call. And as a side effect ringtone is cut.

It doesn't matter whether phone is locked or not, everything looks like Signal doesn't handle the notification itself. I tried all possible options in Android and Signal but the behavior is the same.
1. Signal is added to Never Sleep exceptions, background data usage is allowed
2. DnD is disabled
3. Notifications profiles are not used
4. Issue exists with both WiFi and LTE data connections
5. Sometimes (very rarely) everything works as intended -- full screen accept/decline window appears
6. I also have Skype and other messengers -- they work seamless, full screen call accept/decline window appears so it's Signal problem only
7. Signal app doesn't have "Appear on top" configuration option like Skype does. Maybe it could be one of the reasons?

**Actual result:** Only small short-living Android notification appears for several seconds after that sound and vibration is dropped.
**Expected result:** Full screen Signal Window with caller's photo and green/red accept/decline button appears

### Device info

**Device:** Galaxy S22
**Android version:** 12.0.0
**Signal version:** 6.22.8

### Link to debug log
https://debuglogs.org/android/6.22.8/a25683c98a90ac037b69e6bcd955f9a4993f16bfafbd37c9a653ebc6ed8e1ed8

**Comment by cody-signal** [comment]
Hey there, thanks for the detailed report!

We hope the fix for this is already in 6.23.2 which is in beta right now. Will start to become available in prod in the coming days.

**Comment by eugeneseppel** [comment]
Hello,
6.23.5 is very helpful, thank you! Now the full screen notification and continuous sound appears (as expected) when screen is off (phone is locked). 
However there still is small issue -- if phone is not locked (screen is active) only pop-up notification appears. To see the full screen notification one needs to press on pop-up.
Also there is another big problem -- It's not possible to configure Signal to ignore "Don't Disturb", however I'll create a separate ticket for that

**Comment by Gapva** [comment]
Hello, I am having the same issue where, sometimes, I am unable to answer an incoming call because there is no interactive interface that allows me to actually do so.

This may be unrelated or not relevant to this issue at all, but I have also noticed that Signal is nowhere to be seen in the "Display over other apps" context menu in Android settings.
###### (Note: The reason I say "may be" is because I'm not sure if Signal is even supposed to be in said context menu at all.)

**Comment by cody-signal** [comment]
@Gapva can you update to the latest version of Signal and post a new debuglog after encountering the issue?

**Comment by Anonymouss76** [comment]
> @Gapva can you update to the latest version of Signal and post a new debuglog after encountering the issue?

I use Signal 6.26.3 and I have HUGE problems with notification when someone is calling me. My phone freezes for few seconds and then buggy notification come to answer/decline call. But only sometimes. Often times I hear my ringtone but absolutely no sign of any notification! This is happening to Signal only, no other apps is effected. And just after 6.23 for Android.

Samsung Galaxy A52 and other multiple phones.

I checked everything and put Sognal on "no sleep" and "work always" in system setting but the problem is the same. This is Signal only problem and it is complete deal breaker!

**Comment by DubbleClick** [comment]
Similar thing happening for me - when I get called, the phone starts vibrating but no popup appears. The screen is entirely frozen until Signal finally manages to show the accept/decline buttons roughly 10 seconds later.

S23/latest update. Reinstalling did not help.

**Comment by cody-signal** [comment]
Thanks for the reports and sorry you are encountering these issues. We are having trouble reproducing on our end. Could you provide update the debuglogs after encountering the issue on the latest version of Signal available to you?

**Comment by DubbleClick** [comment]
How do you create a debug log?

**Comment by cody-signal** [comment]
Settings->Help->Debuglog

Best thing is to encounter the issue, submit the debuglog, and then post the link here or you can send it to me directly via email. <my first name> at signal dot org.

**Comment by DubbleClick** [comment]
https://debuglogs.org/android/6.29.2/9323be9a62cbb682de4a1fae92c2e662a23fdf5e5e07025b7610ede51fe286aa

I set up Signal again, had someone call me and the issue occured. Ringing started right away, phone froze for 10 seconds, when it finally showed the auI I immediately picked up and that led to "System UI isn't responding".

**Comment by cody-signal** [comment]
Gotcha, thanks! We made a change in 6.30.4 that may help with this problem. When it's available to you to install please try again and let us know if it fairs any better.

**Comment by DubbleClick** [comment]
Will keep you updated once it becomes available.

**Comment by hero2zero** [comment]
Same issue repeatedly on Galaxy s23. Phone freezes on incoming call and persists with blank screen until call stops. Phone is unresponsive during incoming calls despite pressing power button.

**Comment by DubbleClick** [comment]
> Gotcha, thanks! We made a change in 6.30.4 that may help with this problem. When it's available to you to install please try again and let us know if it fairs any better.

I'm on 6.30.4 now and the issues are the exact same. No change.

**Comment by greyson-signal** [comment]
We have another hopeful fix rolling out in 6.31.2. That's in beta and beginning a prod rollout. Let us know if that helps!

**Comment by Bostrolicious** [comment]
> We have another hopeful fix rolling out in 6.31.2. That's in beta and beginning a prod rollout. Let us know if that helps!

I had the same issue, but on a Galaxy S23 on Android 13. I just joined the beta to see if 6.31.2 helps, and it seems like it did! I only tried two calls, but previously all calls resulted in frozen UI.
